src/core/server/saved_objects/service/lib/aggregations/aggs_types/bucket_aggs.ts

```diff
@@ -44,9 +44,24 @@ import { sortOrderSchema } from './common_schemas';
  * - variable_width_histogram
  */
 
+// TODO: this needs to be recursively built somehow
+// note: node_modules/@elastic/elasticsearch/lib/api/typesWithBodyKey.d.ts
+const termSchema = s.object({
+  term: s.recordOf(s.string(), s.oneOf([s.string(), s.boolean(), s.number()])),
+});
+
+// TODO: this needs to be recursively built somehow
+// note: node_modules/@elastic/elasticsearch/lib/api/typesWithBodyKey.d.ts
+const boolSchema = s.object({
+  bool: s.object({
+    must_not: s.oneOf([termSchema]),
+  }),
+});
+
 export const bucketAggsSchemas: Record<string, ObjectType> = {
-  filter: s.object({
-    term: s.recordOf(s.string(), s.oneOf([s.string(), s.boolean(), s.number()])),
+  filter: termSchema,
+  filters: s.object({
+    filters: s.recordOf(s.string(), s.oneOf([termSchema, boolSchema])),
```

Comment:
@rudolf @pgayvallet I'd like to implement the `filters` aggregation. It'd be great if we could build these in some recursive fashion since they can be deeply nested with `bool/filter/term/etc` queries. I believe the `io-ts` library supports recursive types: https://github.com/gcanti/io-ts/blob/master/index.md#recursive-types

Does `kbn/config-schema` support recursive types? Any other suggestions on how to handle this?

Reply:
Based on https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-filters-aggregation.html#_anonymous_filters the second (nested) `filters` can also be an array. Are we gonna leave it for another iteration?

Reply:
@cnasikas good point. I chatted with Rudolf and Pierre and we decided to just implement the minimum that I needed haha. I think once the Kibana Persistence Toolkit is implemented it'll be obsolete anyway.

Reply:
Ok!